util/helper-cairo.cc

```diff
@@ -89,10 +89,16 @@ helper_cairo_create_scaled_font (const font_options_t *font_opts)
       atexit (free_ft_library);
 #endif
     }
-    FT_New_Face (ft_library,
-		 font_opts->font_file,
-		 font_opts->face_index,
-		 &ft_face);
+
+    unsigned int blob_length;
+    const char *blob_data = hb_blob_get_data (font_opts->blob, &blob_length);
```

Comment:
We shouldn't have pass file path directly to the freetype, that was causing shaping with one font and rendering with another, the weirdness you were describing Behdad.

Reply:
Right. Thanks.